src/Juvix/Compiler/Abstract/Translation/FromConcrete.hs

```diff
@@ -336,6 +336,39 @@ goExpression = \case
       ScopedFunction fun -> Abstract.IdenFunction (Abstract.FunctionRef (goName (fun ^. Concrete.functionRefName)))
       ScopedConstructor c -> Abstract.IdenConstructor (Abstract.ConstructorRef (goName (c ^. Concrete.constructorRefName)))
 
+    goLet :: LetBlock 'Scoped -> Sem r Abstract.Let
+    goLet l = do
+      _letExpression <- goExpression (l ^. letExpression)
+      _letClauses <- goLetClauses (l ^. letClauses)
+      return Abstract.Let {..}
+      where
+        goLetClauses :: NonEmpty (LetClause 'Scoped) -> Sem r (NonEmpty Abstract.LetClause)
+        goLetClauses cl =
+          nonEmpty' <$> sequence [Abstract.LetFunDef <$> goSig sig | LetTypeSig sig <- toList cl]
+          where
+            goSig :: TypeSignature 'Scoped -> Sem r Abstract.FunctionDef
+            goSig sig = do
+              _funDefClauses <- getClauses
+              _funDefTypeSig <- goExpression (sig ^. sigType)
+              let _funDefBuiltin = sig ^. sigBuiltin
+                  _funDefTerminating = sig ^. sigTerminating
+                  _funDefName = goSymbol (sig ^. sigName)
+                  _funDefExamples :: [Abstract.Example] = []
+              registerFunction' Abstract.FunctionDef {..}
+              where
+                getClauses :: Sem r (NonEmpty Abstract.FunctionClause)
+                getClauses = do
+                  cls <-
+                    sequence
+                      [ goFunctionClause c | LetFunClause c <- toList cl, sig ^. sigName == c ^. clauseOwnerFunction
+                      ]
+                  case nonEmpty cls of
+                    Nothing ->
+                      throw
+                        ( ErrLacksFunctionClause (LacksFunctionClause sig)
```

Comment:
Add a negative test for this? (Unless there's already one I've missed).

Reply:
done in f8069ba7